druid/src/lens/lens.rs

```diff
@@ -125,6 +125,51 @@ pub trait LensExt<A: ?Sized, B: ?Sized>: Lens<A, B> {
         self.then(Deref)
     }
 
+    /// Invoke a type's `AsRef` and `AsMut` impl.
+    ///
+    /// It also allows indexing arrays with the [`index`] lens as shown in the example.
```

Comment:
maybe worth explaining that this is because `Index` is implemented for slices, and not arrays. Otherwise it's a little mysterious (to me, anyway) why reffing allowing indexing to work

Reply:
Good idea! I've added a note.